### PR TITLE
Implement User controller for /user

### LINKED ISSUE #61: Formalise API spec for User endpoint

### PR DESCRIPTION
**Pull request by ningyuansg**
Not sure if substantial enough to close #61, #62.

### Todos

- Get xp per user (currently using hard-coded `xp: 0`).

### CODE REVIEW

**Comment by ningyuansg** [comment]
Bump. Would like a review and merge so we can implement username and role fetching on frontend by today.

File: lib/cadet_web/controllers/user_controller.ex [hunk 1]
```diff
@@ -2,32 +2,27 @@ defmodule CadetWeb.UserController do
...
   """
-  use CadetWeb, :controller
 
+  use CadetWeb, :controller
   use PhoenixSwagger
 
+  def index(conn, _) do
+    user = conn.assigns.current_user
+    # TODO: fetch total xp for this user
+    render(conn, "user_info.json", name: user.name, role: user.role, xp: 0)
```
Comment (indocomsoft) [inline lib/cadet_web/controllers/user_controller.ex]: I think it's preferable you just pass the user directly and use the controller's method name in the view.
```elixir
render(conn, "index.json", user: user, xp: 0)
```

File: lib/cadet_web/views/user_view.ex [hunk 2]
```diff
@@ -0,0 +1,7 @@
+defmodule CadetWeb.UserView do
+  use CadetWeb, :view
+
+  def render("user_info.json", %{name: name, role: role, xp: xp}) do
```
Comment (indocomsoft) [inline lib/cadet_web/views/user_view.ex]: You should change to `def render("index.json", %{user: user, xp: xp} do`

File: test/cadet_web/controllers/user_controller_test.exs [hunk 3]
```diff
@@ -1,10 +1,27 @@
...
     assert is_map(UserController.swagger_definitions())
     assert is_map(UserController.swagger_path_index(nil))
   end
+
+  describe "GET /user" do
+    test "success", %{conn: conn} do
+      user = insert(:user, %{role: :student})
+      conn = Guardian.Plug.sign_in(conn, user)
+      conn = get(conn, "/v1/user", nil)
+      assert response(conn, 200)
```
Comment (indocomsoft) [inline test/cadet_web/controllers/user_controller_test.exs]: You should test the result as well:
```elixir
body = json_response(conn, 200)
expected = %{"name" => user.name, "role" => "#{user.role}", xp: 0}

assert ^expected = body 
```

File: test/cadet_web/controllers/user_controller_test.exs [hunk 4]
```diff
@@ -1,10 +1,27 @@
...
+    end
+
+    test "unauthorized", %{conn: conn} do
+      conn = get(conn, "/v1/user", nil)
+      assert response(conn, 401)
```
Comment (indocomsoft) [inline test/cadet_web/controllers/user_controller_test.exs]: `assert response(conn, 401) =~ "Unauthorised"`

**Pull request merged**